pkg/agent/controller/egress/egress_controller.go

```diff
@@ -521,17 +522,27 @@ func (c *EgressController) unbindPodEgress(pod, egress string) (string, bool) {
 	return "", false
 }
 
-func (c *EgressController) updateEgressStatus(egress *crdv1a2.Egress, nodeName string) error {
-	if egress.Status.EgressNode == nodeName {
+func (c *EgressController) updateEgressStatus(egressName, nodeName string) error {
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		egress, err := c.crdClient.CrdV1alpha2().Egresses().Get(context.TODO(), egressName, metav1.GetOptions{})
```

Comment:
Normally the Egress in Lister is in sync, it should only get the value from API when an update meets conflict (which means the cache in Lister is out of sync for this resource). Otherwise it would waste one GET request for each update.

Reply:
done